src/types/internal/schema-field.ts

```diff
@@ -0,0 +1,328 @@
+/**
+ * @license
+ * Copyright (c) 2020 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from '../../platform/assert-web.js';
+import {Refinement} from './refiner.js';
+import {EntityType, ReferenceType, Type} from './type.js';
+import {AnnotationRef} from '../../runtime/arcs-types/annotation.js';
+import {SchemaPrimitiveTypeValue, KotlinPrimitiveTypeValue, BinaryExpressionNode} from '../../runtime/manifest-ast-types/manifest-ast-nodes.js';
+
+export enum Kind {
+  Primitive = 'schema-primitive',
+  KotlinPrimitive = 'kotlin-primitive',
+  Collection = 'schema-collection',
+  Reference = 'schema-reference',
+  OrderedList = 'schema-ordered-list',
+  Union = 'schema-union',
+  Tuple = 'schema-tuple',
+  Nested = 'schema-nested',
+  Inline = 'schema-inline',
+  TypeName = 'type-name' // same as inline.
+}
+
+// tslint:disable-next-line: no-any
+type SchemaFieldMethod  = (field: {}) => SchemaField;
+
+export abstract class SchemaField {
```

Comment:
SchemaFieldType or SchemaType, I think.

Reply:
renamed to SchemaFieldType.
actually it is FieldType in KT, maybe better to make them consistent, so maybe FieldType?

Reply:
FieldType is good.

Reply:
done